core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -57,15 +57,21 @@
   public <T> T newInstance(Target<T> target) {
     Map<String, MethodHandler> nameToHandler = targetToHandlersByName.apply(target);
     Map<Method, MethodHandler> methodToHandler = new LinkedHashMap<Method, MethodHandler>();
+    Promise<Object> proxyPromise = new Promise<Object>();
+
     for (Method method : target.type().getMethods()) {
       if (method.getDeclaringClass() == Object.class) {
         continue;
+      } else if(Util.isDefault(method)) {
+        methodToHandler.put(method, new DefaultMethodHandler(method, proxyPromise));
```

Comment:
here's another idea.. since this is mutable, skip the promise/ref thing. after you create the handler, loop through methodToHandler and bind the proxy before returning. this should be more thread safe.

ex. 
```java

boolean supportsDefault = false;
for (Method method : ...
...
} else if(Util.isDefault(method)) {
  supportsDefault = true;
...
T proxy = ...
if (supportsDefault) { // guards against execution in old JREs
  for (MethodHandler handler: methodToHandler.values()) {
    if (handler instanceof DefaultMethodHandler) {
      ((DefaultMethodHandler)handler).bindTo(proxy);
    }
  }
}
```